### PR TITLE
LOG-1814: Avoid use deprecated methods in Cluster Logging Operator

### PR DESCRIPTION
**Pull request by Clee2691**
### Description
This PR unifies fake client creation utilizing `NewFakeClient` in lieu of `NewClientBuilder`.

/cc @vparfonov @cahartma 
/assign @jcantrill

### Links
- JIRA: https://issues.redhat.com/browse/LOG-1814

### CODE REVIEW

**Comment by xperimental** [comment]
fwiw: The `NewFakeClient` function was only deprecated in a few versions of `controller-runtime`. We've updated to a newer version where it's not deprecated anymore, so just removing `//nolint` from the functions is enough, the call does not need to be replaced.

**Review by alanconway — APPROVED** [review]
+1  The Builder does provide extra flexibility for more complicated tests (e.g. if you need to control the Schema being used etc.) but there's no need to use it when its not required, which is most of the time.

**Review by jcantrill — COMMENTED** [review]
/lgtm

File: internal/reconcile/scc_test.go [hunk 1]
```diff
@@ -36,11 +37,11 @@ var _ = Describe("reconciling ", func() {
 
 		globalScheme := scheme.Scheme
 		Expect(security.Install(globalScheme)).To(Succeed())
-		builder := fake.NewClientBuilder().WithScheme(globalScheme)
+
```
Comment (xperimental) [inline internal/reconcile/scc_test.go:40]: If you want, you could also inline `globalScheme` above into the `Install()` call. It's just a fancy `AddToScheme` :slightly_smiling_face:

**Review by xperimental — COMMENTED** [review]
lgtm

**Pull request merged**